upgrade.go

```diff
@@ -0,0 +1,231 @@
+package upgrade
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
+	"github.com/termie/go-shutil"
+)
+
+func CurrentVersion(version string) (semver.Version, error) {
+	return semver.Make(version)
+}
+
+func AvailableVersion(repo string, filePrefix string, channel string) (semver.Version, error) {
+	resp, err := http.Get(
+		fmt.Sprintf("%s/%s%s", repo, filePrefix, channel),
+	)
+	if err != nil {
+		return semver.Version{}, err
```

Comment:
Convention question, why return this type over returning nil?

Reply:
The return type of the function is a struct, not a pointer to a struct. It is probably more idiomatic to return a pointer, but there's no requirement for it. In Go, nil is the zero value for pointers, interfaces, maps, slices, channels and function types, representing an uninitialized value. (https://stackoverflow.com/a/35983594)